lib/fog/kubevirt/compute/models/services.rb

```diff
@@ -0,0 +1,81 @@
+require 'fog/core/collection'
+require 'fog/kubevirt/compute/models/service'
+
+module Fog
+  module Kubevirt
+    class Compute
+      class Services < Fog::Collection
+        attr_reader :kind, :resource_version
+
+        model Fog::Kubevirt::Compute::Service
+
+        def all(filters = {})
+          begin
+            srvs = service.list_services(filters)
+          rescue ::Fog::Kubevirt::Errors::ClientError
+            # we assume that we get 404
+            srvs = []
+          end
+          @kind = srvs.kind
+          @resource_version = srvs.resource_version
+          load srvs
+        end
+
+        def get(name)
+          new service.get_service(name)
+        end
+
+        # Creates a service using provided paramters:
+        # :name [String] - name of a service
+        # :port [int] - a port which will be exposed on node and cluster
+        # :target_port [int] - a vmi port which will be forwarded
+        # :vmi_name [String] - name of a vmi to be selected
+        #
+        # @param [Hash] attributes containing details about service to be
+        #   created.
+        def create(args = {})
+          port = args[:port]
+          name = args[:name]
+          target_port = args[:target_port]
+          vmi_name = args[:vmi_name]
+
+          srv = {
+            :apiVersion => "v1",
+            :kind => "Service",
+            :metadata => {
+              :name      => name,
+              :namespace => service.namespace
+            },
+            :spec => {
+              :externalTrafficPolicy => "Cluster",
+              :ports => [
+                {:nodePort   => port,
+                 :port       => port,
+                 :protocol   => "TCP",
+                 :targetPort => target_port
+                }
+              ],
+              :selector => {
+                :special => vmi_name
+              },
+              :type => "NodePort"
```

Comment:
Do we want to allow creation of a single type of service ? wouldn't the type should be passed as an argument and set "NodePort" as the default if none was provided (even-though the default on kubernetes is clusterP) ?

Reply:
Good idea. Will accept type as parameter